src/repositories/schemas.ts

```diff
@@ -1,9 +1,183 @@
 import { OAuth1Client, ObjectId } from '@extrahorizon/javascript-sdk';
 
+interface Transition {
+  name: string;
+  type: string;
+  fromStatuses: string[];
+  toStatus: string;
+  conditions: { type: string; };
+  actions: { type: string; };
+  afterActions: { type: string; };
+}
+
 export async function remove(sdk: OAuth1Client, schemaId: ObjectId) {
   return await sdk.data.schemas.remove(schemaId);
 }
 
 export async function disable(sdk: OAuth1Client, schemaId: ObjectId) {
   return await sdk.data.schemas.disable(schemaId);
 }
+
+/**
+ * fetch a full schema from the service from a name
+ * @param {string} name the name of a schema
+ */
+export async function fetchSchemaByName(sdk: OAuth1Client, name: string) {
```

Comment:
Apart from adding sdk as the first argument these function have been unchanged straight copies from the dataService that is removed.

Reply:
Also in the last commit I replace directly returning the responses from the raw calls with `response.data` to make the mocking cleaner. Responses of those copied functions are mostly unused anyway.